kiva/Auth0.Android
# Provide migration path for unique key alias pattern

Migrates users who are on the legacy key alias pattern to the new unique pattern. In order to better test and debug these changes, I integrated both Auth0 and Lock as submodules. When this work is approved, I will either include these as .aars or add the libraries to github packages. The androidx migration was required to get this to build as a submodule within the main project.

+ On **decryption** only, uses legacy keys if present
+ Throws `NeedsMigrationException` so that `SecureCredentialsManager` can re-save the credentials
+ All calls to `encrypt` use the new, unique key alias pattern
+ Synchronizes call to get credentials. This prevents encrypt/decrypt collisions
+ Refactors some of the keystore logic into a helper class to make `CryptoUtil` a little bit easier to read

Test and debug the migration path by checking out branch `MOB-1093-auth0-migration` in the Kiva project. You will need to clean and rebuild. 

1. Download the latest apk from app distribution
2. Login
3. Close the app
4. Run `MOB-1093-auth0-migration`
5. Verify you are still logged in. Validate the keystore keys by using the debugger
